crates/gitbutler-project-store/src/migrations/migrator.rs

```diff
@@ -0,0 +1,204 @@
+                        (Uuid::new_v4(), migration.name),
+                    )
+                    .context("Failed to insert migration completion marker")?;
+
+                transaction.commit()?;
+            }
+        }
+
+        Ok(())
+    }
+
+    /// Queries all the applied migrations.
+    /// If the `migrations` table doesn't exist, it will be created here
+    fn find_applied_migrations(&self) -> Result<Vec<String>> {
+        self.connection
+            .execute(CREATE_MIGRATIONS_TABLE, [])
+            .context("Failed to create migrations table")?;
+
+        let mut statement = self
+            .connection
+            .prepare("SELECT name FROM migrations")
+            .context("Failed to fetch migrations")?;
+
+        let mapped_rows = statement
+            .query_map([], |row| row.get(0))?
+            .collect::<Result<Vec<_>, _>>()?;
+
+        Ok(mapped_rows)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn first_migration() -> Migration {
+        Migration {
+            name: "first".to_string(),
+            up: |connection| {
+                connection.execute(
+                    "
+                CREATE TABLE testaroni (
+                    id BLOB PRIMARY KEY, -- A V4 UUID
+                    name TEXT NOT NULL
+                )
+                ",
+                    [],
+                )?;
+
+                Ok(())
+            },
+        }
+    }
+
+    /// Depends on `first_migration`
+    fn second_migration() -> Migration {
+        Migration {
+            name: "second".to_string(),
+            up: |connection| {
+                connection.execute("ALTER TABLE testaroni ADD potatoes TEXT", [])?;
+
+                Ok(())
+            },
+        }
+    }
+
+    #[test]
+    /// Testing the `find_applied_migrations` function.
+    /// Ensuring that it creates the `migrations` table if it doesn't exist
+    fn find_applied_migrations_creates_migrations_table_if_not_exists() {
+        let mut connection = Connection::open_in_memory().unwrap();
+
+        let migrator = Migrator::new(&mut connection);
+
+        // Try to find unapplied migrations
+        migrator.find_applied_migrations().unwrap();
+
+        let statement = connection
+            .prepare("SELECT name FROM sqlite_master WHERE type='table' AND name='migrations'")
+            .unwrap();
+        assert_eq!(statement.column_count(), 1);
+    }
+
+    #[test]
+    /// Testing the `find_applied_migrations` function.
+    /// Assuming the `migrations` table exists (which we create by calling
+    /// `find_applied_migrations), we expect that any values inserted into
+    /// the table will get returned when we call it again.
+    fn find_applied_migrations_lists_all_entries_in_table() {
+        let mut connection = Connection::open_in_memory().unwrap();
+
+        // Call find_applied_migrations in order to create the migrations table
```

Comment:
I'd move this comment into an assertion as last argument of `assert!` for example:

```Rust
assert!(migrator.find_applied_migrations().unwrap().is_empty(), "nothing happened yet, but creates table as side-effect");
```